### ISSUE TITLE
Ogg Vorbis track does insert delay and/or artifacts entering a loop

### ISSUE THREAD

**Issue by smesgr9000** [mixxxdj/mixxx#11267]
### Bug Description

this issue is not reproducable for all files/songs, but if a file has this issue it is reproducable at 100% for me. As far as I'm aware this only happens with ogg vorbis. Didn't notice this for mp3s or wavs.

Way to reproduce and easily notice it.
# convert the ogg file to a wav, but keep ogg
# Start Mixxx
# load ogg in one deck, the same file as a wav in another deck
# set loop at some point mid track on ogg file (or have it already set in database)
# start both in parallel and wait for the loop entry (which don't need to be active)

example database entry for [this](https://www.youtube.com/watch?v=eEHVSh6ZJPw) song:
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head><title></title><meta name="generator" content="DB Browser for SQLite"><meta name="date" content="YYYY-02-DDT11:17:30.573"><style type="text/css">br{mso-data-placement:same-cell;}</style></head><body>

112 | 4559 | 2 | 3160 | 0 | -1 |   | 16744448
-- | -- | -- | -- | -- | -- | -- | --
113 | 4559 | 8 | 3160 | 41131816 | -1 |   | 16744448
114 | 4559 | 6 | 3160 | 0 | -1 |   | 16744448
115 | 4559 | 7 | -1 | 41134976 | -1 |   | 16744448
388 | 4559 | 1 | 37858326.9677419 | 0 | 6 |   | 16744448
447 | 4559 | 4 | 2692133 | 341419.35483871 | -1 |   | 16744448
524 | 4559 | 1 | 9733907.61290323 | 0 | 3 | Gesang | 16744448
525 | 4559 | 1 | 3456 | 0 | 0 | Ü Start | 16744448
527 | 4559 | 1 | 2692133.41935484 | 0 | 1 |   | 16744448

</body></html>

I did try changing the audio buffer and settings but does not mitigate the issue - the resulting effect maybe differ but still affected.

Attached a short sample of the steps above
[delay.mp3.zip](https://github.com/mixxxdj/mixxx/files/10713160/delay.mp3.zip)

### Version

2.3.2

### OS

Ubuntu 22.04

**Comment by daschuer** [comment]
I do not understand what the issue is. Looping works on the decoded raw samples, so an issue cannot be related to looping. 
There may be an issue with random seeks or a general decoding issue.  

Can you describe again what happens and what should happen instead? 
Can you provide the wav and the ogg file as a private message?

**Comment by smesgr9000** [comment]
sorry it is hard to explain. Hope this helps:
![issue](https://user-images.githubusercontent.com/90571089/218275248-b9fc7e43-1687-4e2a-9fff-f3fdc80811ed.png)

- left deck has WAV
- right deck has OGG
- just start playback of both decks (default by pressing "D" and "L" at the same time) and wait
- music plays
- at reaching the Hotcue point 2 and loop entry the OGG playback is suddenly shifted. The effect is dependent of the track, sometimes it looks the currently played not is played twice than continue playing.

expectation: in this setup the music continue to play as "one" song to the end.
seen: the playback on both files seem to be shifted by some milliseconds at hotcue point 2

**Comment by daschuer** [comment]
Ah, now I understand. Mixxx keeps the portions of samples in cache that are likely to be played next, to avoid tiny gaps of silence for accessing the disk. When you reach a cue point, it uses the already cached samples instead of the samples from a steady stream. This should not make differences. Since it makes a difference for you, the ogg decoder id probably broken.

**Comment by daschuer** [comment]
I can confirm this with Ubuntu Focal 20.4

**Comment by daschuer** [comment]
When I load the original ogg file with Audacity and then export it as new ogg file, the issue is gone. 
So I think it is something in the original file that tricks Mixxx. Unlike mp3 where we manage the seeking yourself we rely for ogg on libvorbisfile. So we may consider to file an upstream bug at: 
https://github.com/xiph/vorbis/issues
Do you have more files with this issue?

**Comment by smesgr9000** [comment]
yes, but can't pin-point it at the moment. Of course most of them are generated by the same lib

**Comment by daschuer** [comment]
I have just compared the original and the Audacity re-encoded file with oggz-info

With this result:

Original 
```
Content-Duration: 00:07:46.382

Vorbis: serialno 1179750276
	32655 packets in 464 pages, 70.4 packets/page, 0.871% Ogg overhead
	Audio-Samplerate: 44100 Hz
	Audio-Channels: 2
```
re-encoded:
```
Content-Duration: 00:07:46.419

Vorbis: serialno 1478221743
	39067 packets in 1735 pages, 22.5 packets/page, 1.387% Ogg overhead
	Audio-Samplerate: 44100 Hz
	Audio-Channels: 2
```
What does  packets/page mean?

**Comment by smesgr9000** [comment]
odd the duration differs - this shouldn't happen right?

**Comment by daschuer** [comment]
We may consider to add an automatic test to detect such issues and warn the user. We have already have a check decoding offsets at the beginning. we may extend that for the end as well.
